src/Passes/Source/Rules/OperandPrototype.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Rules/OperandPrototype.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    IOperandPrototype::~IOperandPrototype() = default;
+    bool IOperandPrototype::matchChildren(Value* value, Captures& captures) const
+    {
+        auto user = llvm::dyn_cast<llvm::User>(value);
+        if (!children_.empty())
+        {
+            if (user == nullptr)
+            {
+                return false;
+            }
+
+            if (user->getNumOperands() != children_.size())
+            {
+                return false;
+            }
+
+            uint64_t i = 0;
+            while (i < children_.size())
+            {
+                auto v = user->getOperand(static_cast<uint32_t>(i));
+                if (!children_[i]->match(v, captures))
+                {
+                    return false;
+                }
+                ++i;
+            }
+
+
+            return true;
+        }
+
+        //  llvm::errs() << "SUCCESS MATCH: " << *value << " " << user->getNumOperands() << "\n";
+        // TODO(tfr): Check other possibilities for value
```

Comment:
It would be good to track todo's in the code with tasks in ADO (usually GitHub, but given that this code base is still very much in flux, just a quick note in ADO makes more sense).
Same comment as above regarding control prints and commented out code.

Reply:
Yes, I agree. I haven't been able to do this until now as I had no place to do so.